pkg/controller/master/setting/ntp_server.go

```diff
@@ -0,0 +1,90 @@
+package setting
+
+import (
+	"reflect"
+	"strings"
+
+	nodev1 "github.com/harvester/node-manager/pkg/apis/node.harvesterhci.io/v1beta1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+)
+
+const (
+	nodeConfigNS = "harvester-system"
+)
+
+func (h *Handler) syncNTPServer(setting *harvesterv1.Setting) error {
+	logrus.Infof("Processing setting %s, value: %s", setting.Name, setting.Value)
+
+	ntpServers := strings.Replace(setting.Value, ",", " ", -1)
+	if ntpServers == "" {
+		logrus.Infof("NTP server is empty, do no-op here.")
+		return nil
+	}
+
+	nodes, err := h.nodeClient.List(metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("List all node failure: %v", err)
+		return err
+	}
+
+	for _, node := range nodes.Items {
+		nodeName := node.Name
+		nodeConfig, err := h.nodeConfigs.Get(nodeConfigNS, nodeName, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			logrus.Errorf("Get node config with nodeName %s failure: %v", nodeName, err)
+			return err
+		}
+		logrus.Debugf("Nodeconfig content: %+v, notFound: %v", nodeConfig, errors.IsNotFound(err))
+		if nodeConfig == nil || errors.IsNotFound(err) {
+			nodeConfig := generateNodeConfig(nodeName, ntpServers)
+			_, err := h.nodeConfigs.Create(nodeConfig)
+			if err != nil {
+				logrus.Errorf("Create node config with nodeName %s failure: %v", nodeName, err)
+			}
+		} else {
+			nodeConfigCpy := nodeConfig.DeepCopy()
+			if nodeConfigCpy.Spec.NTPConfig == nil {
+				nodeConfigCpy.Spec.NTPConfig = generateNTPConfig(ntpServers)
+			} else {
+				nodeConfigCpy.Spec.NTPConfig.NTPServers = ntpServers
+				nodeConfigCpy.Spec.NTPConfig.ConfigModified = true
+			}
```

Comment:
why not make `nodeConfigCpy.Spec.NTPConfig = generateNTPConfig(ntpServers)` the default behaviour?

Reply:
That is a good suggestion. I will rewrite this.  Thanks!